module_vd_gas_wesely.f90

```diff
@@ -0,0 +1,211 @@
+        real, intent(in)    :: henso2
+        real, intent(in)    :: f0
+        real, intent(in)    :: rscale
+        real, intent(in)    :: ts
+        real, intent(in)    :: dstress(0:5)
+        real*8, intent(in)  :: solflux ! why does this one have precision but the others dont? 
+        real, intent(in)    :: ri
+        real, intent(in)    :: rlu
+        real, intent(in)    :: rac
+        real, intent(in)    :: rlcs
+        real, intent(in)    :: rlco
+        real, intent(in)    :: rgss
+        real, intent(in)    :: rgso
+        real, intent(in)    :: vd
+
+        !Step 1. Calculate atmospheric resistance (due to turbulence diffusion), Ra
+        ra = (log(deltaz/z0)-psih)/(vk * ustar)
+        ra = amax1(ra,rmin)
+
+        !Step 2. Calulate deposition layer resistance (due to molecular diffusion), Rd
+        schmidt = vair * diffrat/diffh2o
+        rd = d1 * schmidt**d2/(vk * ustar)
+        rd = amax1(rd,rmin)
+
+        !Step 3. Calculate surface layer resistance, Rs
+        !!Step 3-1. special case surface layer over water
+        if (ilu == 7) then
+            if (io3 == 1 .and. icdocn == 1) then
+                rs = 1./(1.e-4 + 5.e-6 * henry *ustar * ts **3)                                   !Follow Helmig et al (2012)
+                rs = amax1(rs,1500.)
+            else
+                rs = 1./ (3.9e-5*henry*ustar*(ts+273.14))                                         !Follow Kumar et al (1996) and Sehmel (1980)
+                rs = amax1(rs,rmin)
+            endif
+            goto 100 ! need to remove goto blocks - they are not accepted by NCO
+        endif 
+
+        !!Step 3-2. normal case, surface layer over land 
+        !!!Step 3-2-1.stomatal and mesophyll resistance,Rst & Rm
+        rst = rmax
+        if (ts>0 .and. ts<40) then
+            rst = diffrat*ri*(1.+(200./(solflux + 0.1))**2)*(400./(ts*(40.-ts)))  !Follow Wesely et al.,(1989) eqn (3)&(4)   
+            rst = rst * dstress(istress)
+        endif
+        !Effect of dew and rain,increased by a factor of 3, as leafs are covered
+        if (iwet >0) then
+            rst = 3.* rst                                                                         
+        endif
+        
+        rst = amin1(rmax,rst)
+
+        !Follow Wesely et al., (1989) eqn (6)
+        rm = 1./(henry/3000.+100.*f0)                                                             
+        rm = amax1(rmin,rm)
+        rm = amin1(rmax,rm)
+
+    !   !!Step 3-2-2.upper canopy resistance,Ruc
+        if (iwet == 0) then                     !dry surface
+            ruc = rlu/(henry/henso2+f0)
+            ruc = amin1(rmax,ruc)
+        else
+            if (iwet == 1) then                 !dew surface
+                rlus = 100.
+                if (ilu == 1) then
+                    rlus = 50.
+                endif
+    !            rluo = 1./(1./3000.+1./(3.*rlu))                                                   !Follow Wesely et al.,(1989) eqn (11)
```

Comment:
another question is why in CAMx code they turn off wesely formula, and used simple 1000+rlu formula instead here?

Reply:
I'm not sure.  maybe rick will know a better answer